src/screens/TransactionDetail.js

```diff
@@ -94,6 +94,25 @@ class TransactionDetail extends React.Component {
 
     try {
       const data = await this.props.wallet.getFullTxById(this.props.match.params.id);
+      for (const output of data.tx.outputs) {
+        if (!output.token) {
+          if (output.token_data === 0) {
+            output.token = hathorLib.constants.HATHOR_TOKEN_CONFIG.uid;
+          } else {
+            output.token = data.tx.tokens[(output.token_data & hathorLib.constants.TOKEN_INDEX_MASK) -1].uid;
+          }
+        }
+      }
+
+      for (const input of data.tx.inputs) {
+        if (!input.token) {
+          if (input.token_data === 0) {
+            input.token = hathorLib.constants.HATHOR_TOKEN_CONFIG.uid;
+          } else {
+            input.token = data.tx.tokens[(input.token_data & hathorLib.constants.TOKEN_INDEX_MASK) -1].uid;
+          }
+        }
+      }
```

Comment:
`getFullTxById` fetches the transaction from the api, which uses has a slightly different structure from the websocket transactions and transactions received via history sync.
These lines just add the `token` uid where the helper will need it.

The issue here was that not having the `token` would make the transaction balance think that there was only a single token of uid `undefined`.

Reply:
I'm not sure I get this: would we have some problem if we implemented this in the lib itself, so the `getFullTxById` method always returns with the right token?

I mean, when using this method in other wallets, won't they need to perform a similar treatment to this response?

Reply:
You are absolutely right, the only other place this is used is on the decode api of the headless and when it is used the token would be missing. I created a PR to fix this on the headless https://github.com/HathorNetwork/hathor-wallet-headless/pull/342

The main problem in my opinion is that the same transaction can have multiple structures depending on which api is used to fetch it from the fullnode or if it comes from the websocket.
Maybe the best solution would be to normalize the transaction model sent by the fullnode and the wallet-lib would be more stable.
Since solving this on the wallets is the simpler approach I think we should stick with it for now.
What do you think?

Reply:
I think it's ok to stick to the simpler approach. But we should have an issue to implement the best solution in the future.